src/core/positioning/egenioussreceiver.cpp

```diff
@@ -0,0 +1,124 @@
+#include "egenioussreceiver.h"
+
+#include <QHostAddress>
+#include <QJsonDocument>
+#include <QJsonValue>
+
+EgenioussReceiver::EgenioussReceiver( QObject *parent )
+  : AbstractGnssReceiver( parent ), mTcpSocket( new QTcpSocket() )
+{
+  connect( mTcpSocket, &QTcpSocket::readyRead, this, &EgenioussReceiver::onReadyRead );
+  connect( mTcpSocket, &QTcpSocket::errorOccurred, this, &EgenioussReceiver::handleError );
+  connect( mTcpSocket, &QTcpSocket::connected, this, &EgenioussReceiver::connected );
+  connect( mTcpSocket, &QTcpSocket::disconnected, this, &EgenioussReceiver::disconnected );
+}
+
+void EgenioussReceiver::handleConnectDevice()
+{
+  mTcpSocket->connectToHost( mAddress, mPort, QTcpSocket::ReadWrite );
+}
+
+void EgenioussReceiver::connected()
+{
+  mSocketState = QAbstractSocket::ConnectedState;
+  emit socketStateChanged( mSocketState );
+  setValid( true );
+}
+
+void EgenioussReceiver::handleDisconnectDevice()
+{
+  mTcpSocket->disconnectFromHost();
+}
+
+void EgenioussReceiver::disconnected()
+{
+  if ( mTcpSocket->state() == QAbstractSocket::ConnectedState )
+  {
+    mSocketState = QAbstractSocket::UnconnectedState;
+    emit socketStateChanged( mSocketState );
+  }
+}
+
+QList<QPair<QString, QVariant>> EgenioussReceiver::details()
+{
+  QList<QPair<QString, QVariant>> detailsList;
+
+  if ( mPayload.isEmpty() )
+  {
+    return detailsList;
+  }
+
+  detailsList.append( qMakePair( "q", mPayload.value( "q" ).toDouble() ) );
```

Comment:
For what stands `q`? Is a key in the json? Eventually to be put in a const to avoid having literals around

Reply:
Yes, it is in JSON and we assume it stands for `Quality(?!)` however in my tests it was always zero.    

if we are sure it is quality i can go for something like this: 

```C++
const QString KEY_QUALITY = "q";
detailsList.append(qMakePair(KEY_QUALITY, mPayload.value(KEY_QUALITY).toDouble()));
```

@nirvn this `KEY_QUALITY` or `KeyQuality` or `keyQuality` ?

Reply:
At this moment let's leave it as is, that's merely to have an extra detail in. We'll polish this when the protocol progresses and we have actual values to handle.